cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java

```diff
@@ -292,4 +279,27 @@ public void getIdleRegions(HttpRequest request, HttpResponder responder,
       LOG.debug("Exception while trying to fetch the idle regions.", e);
     }
   }
+
+  private void initializePruningDebug() {
+    if (debugClazz == null || debugObject == null) {
+      try {
+        this.debugClazz = getClass().getClassLoader()
+          .loadClass("org.apache.tephra.hbase.txprune.InvalidListPruningDebug");
+        this.debugObject = debugClazz.newInstance();
+        Configuration hConf = new Configuration();
+        for (Map.Entry<String, String> entry : cConf) {
+          hConf.set(entry.getKey(), entry.getValue());
+        }
+        Method initMethod = debugClazz.getMethod("initialize", Configuration.class);
```

Comment:
When looking back at Tephra's InvalidListPruningDebug class, I notice that the connection created in intialize is not closed. We need to add a destroy to InvalidListPruningDebug class, and call that method in destroy of this class.